core/dbt/cli/main.py

```diff
@@ -319,7 +319,6 @@ def compile(ctx, **kwargs):
 @p.show_limit
 @p.indirect_selection
 @p.introspect
-@p.parse_only
```

Comment:
@aranke can you confirm that you don't expect or need commands like the following?

```shell
dbt show --parse-only ...
```

I believe the `--parse-only` was originally part of `dbt compile`, and we were planning on removing it completely for Core v1.5.